### ISSUE TITLE
Endpoint base url inconsistency

### ISSUE THREAD

**Issue by friedow** [bptlab/chimera#98]
The base url for the event endpoint group inside the swagger documentation is incorrect. The endpoints are located under the base url `/chimera/api/eventdispatcher`.  This leaves the question wether to correct the documentation or to move the endpoints for the reason of consistency. **What do you think?**

![image](https://user-images.githubusercontent.com/17351844/41441439-ddce54c2-7032-11e8-8b72-4abd00ac37b3.png)

**Comment by marhew** [comment]
I don't know the reason for the separated base URLs and personally don't like it. I would much rather have a short base URL, also dropping the version, like `/api`. Everything else should be then structured under the case model and case, e.g. `/api/casemodel/123/case/423/activity/34/`, also integrating the "history" ( `/api/casemodel/123/case/423/log`).

What I am unsure about **Open Questions** is a) whether we should keep the UUIDs in the API or go back to simple incremented numbers?, and b) whether resources should be named in singular (`case/423`) or plural (`case**s**/423`)? For b) there probably is an existing convention about this matter.

**Please add your opinion**

**Comment by tommartensen** [comment]
TL;DR: 
* Keep the version (to deal with breaking changes in the future)
* some conventions on naming
* What is wrong with UUIDs? Why would you want to change that? What is the higher value of using incremented numbers? (If it exists!)

[API Versioning](https://dev.to/sparkpost/restful-api-versioning-best-practices-why-v1-is-1)
[Conventions](https://restfulapi.net/resource-naming/)
I guess there are many more resources on REST conventions, but this one seemed reasonable enough.

**Comment by julianweise** [comment]
I would suggest to keep versioning having the same reasons in mind @friedow and @tommartensen already stated.
As far as I know "plural naming convention" is the way to go. Usually, you can access all entities of an model by calling `api/entities/` (all => plural 's') and get a single one by calling `api/entities/<entity-id>` (single one => get a specific one identified by id from the overall collection of "entities").

Personally, I prefere simple ID's over UUID's due to the better readability. Beside this, I do not see any other good reason not to choose UUID's.

**Comment by tommartensen** [comment]
I agree with both of you on the readability issue, but there does not seem to be a consensus regarding performance, storage, ... Some people added that you don't want to expose your primary keys via API (security), so it seems like a bunch of questions regarding this.
Keeping in mind that debugging is quite cumbersome in the current state of the application, I would opt for incremental IDs, which may require touching all database tables.

**Comment by janwestphal306** [comment]
Thanks for the great input.
I agree that versioning is a good idea. However, when reaching one of the next milestones we want to make a renaming in the api that is overdue and e.g. use `casemodels` instead of `scenario`. Therefore i would suggest to simply use `api` as the basepath because third party projects like bookshelf need to update the requests paths either way. As for future refactoring of the Chimera api I would then suggest to use `api/v2`, etc..

The reason why we used the UUID instead of incremental ID is because it worked just fine without a database representation. If we wanted to keep track of the number of casemodels ourselves, it could easily lead to inconsistencies. So UUID is regarding the backend as simple as it can be. I know it is a annoying for debugging but the ids are always provided by other rest calls.

**Comment by tommartensen** [comment]
@janwestphal306 I don't quite get, why you would want to keep track of the number of CaseModels (or any other entity), since you can use auto-increment integers in the database, so the application is not responsible for that!

**Comment by janwestphal306** [comment]
@tommartensen I said that because in the beginning of the refactoring we did not use a database, so we could not use auto-increment and needed another way to access the casemodels.

**Comment by tommartensen** [comment]
Interesting. So, I would propose to migrate all existing API endpoints (and their usages) to `api/v2` and keep the UUIDs (so only a minimal renaming is needed) and use `api/v3` for the new and already refactored API endpoints, where we could discuss using UUIDs or incremental ids. 
I would also suggest to create a REST convention document for things like plural 's', trailing slashes, ...

**Comment by marhew** [comment]
Thanks for all the great input. I will include your suggestions into issue #99
